[CDAP-17549] Revamping Pipeline comments

ajainarayanan opened the pull request:
JIRA - https://cdap.atlassian.net/browse/CDAP-17549

This is the base branch to merge all pipeline comments related changes

Have broken down the changes as,

- New components needed for comments - https://github.com/cdapio/cdap/pull/13111
- Integrating comment to angular pipeline - https://github.com/cdapio/cdap/pull/13112
- Changes needed in other components to support pipeline comments - https://github.com/cdapio/cdap/pull/13113
- Support viewing/editing existing pipeline level comments - https://github.com/cdapio/cdap/pull/13129
- E2e tests for pipeline comments - coming soon.

**Context**
This PR is intended to revamp the pipeline comments. What this means is,
  - We will tie comments to specific contexts. In the case of pipeline it is plugins
  - In addition to having plugin comments, we will support existing pipeline level comments
  - Allow users to add comments to a plugin from canvas as well as plugin configuration modal
  - Allow users to view comments in specific context in a published pipeline view

ajainarayanan committed:
Adds base Comments component to be reused in pipelines

  - Adds DynamicAnchoredComments to be able to add comments directly to plugins in pipeline canvas
  - Adds CommentBox, Icon and Menu to hold a single comment
  - Adds comment toggle button to show/hide comments in pipeline

ajainarayanan committed:
Removes existing global (pipeline level) comments and adds plugin level comments support to my-dag directive &
Fixes hydrator modal header styling based on html structure change

ajainarayanan committed:
Modifies Markdown to use useStyles instead of withStyles + changes places where it is used &
Introduces new colors for backward compatibility (when using with bootstrap)

njbriggs commented on cdap-ui/server/config/development/cdap.json:
```diff
@@ -7,7 +7,7 @@
   "dashboard.ssl.disable.cert.check": "false",
   "ssl.external.enabled": "false",
   "router.server.address": "127.0.0.1",
-  "dashboard.bind.port": "11011",
```
Do we want to commit this? (I'm fine with it, BTW - it can be useful to have the built version and the local changes running at the same time.)

ajainarayanan replied:
No this is a placeholder change to have the PR for base branch. All other PRs related to pipeline comments gets merged to this branch and finally we merge this PR to develop. 

But to your point yeah it would be useful to start UI server for development at a different port so that we can compare between SDK and new changes (just like how we do in gitpod environment)

ajainarayanan committed:
Adds pipeline level comments + handles pipelines in deployed view

  - Adds PipelineComments action btn + pipeline level comments to list all global comments for the pipeline
  - Fixes angular to show pipeline level comments in canvas and detailed view
  - Fixes comment button to highlight if there is an existing comment + modifies comment in published pipeline view
